frontend/js/src/explore/lb-radio/LBRadio.tsx

```diff
@@ -0,0 +1,284 @@
+                  target="_blank"
+                  rel="noopener noreferrer"
+                >
+                  {playlistTitle}
+                </a>
+                Heads up: the new playlist is public on Spotify.
+              </>
+            }
+          />,
+          { toastId: "saved-playlist" }
+        );
+      } else {
+        const { error } = await request.json();
+        toast.error(
+          <ToastMsg
+            title="Error"
+            message={`Failed to save playlist to Spotify: ${error}.`}
+          />,
+          { toastId: "saved-playlist-error" }
+        );
+      }
+    } catch (error) {
+      toast.error(
+        <ToastMsg
+          title="Error"
+          message={`Failed to save playlist to Spotify: ${error}.`}
+        />,
+        { toastId: "saved-playlist-error" }
+      );
+    }
+  }, [jspfPlaylist, APIService.APIBaseURI, authToken]);
+
+  const onExportJSPF = React.useCallback(async () => {
+    const jspf = new Blob([JSON.stringify(jspfPlaylist)], {
+      type: "application/json;charset=utf-8",
+    });
+    saveAs(jspf, `${title}.jspf`);
+  }, [jspfPlaylist, title]);
+
+  return (
+    <>
+      <div className="row">
+        <div className="col-sm-12">
+          <Prompt
+            onGenerate={generatePlaylistCallback}
+            errorMessage={errorMessage}
+            initPrompt={promptArg}
+            initMode={modeArg}
+          />
+          <Loader
+            isLoading={isLoading}
+            loaderText="Generating playlist…"
+            className="playlist-loader"
+          >
+            <UserFeedback feedback={feedback} />
+            <Playlist
+              playlist={jspfPlaylist?.playlist}
+              title={title}
+              onSavePlaylist={onSavePlaylist}
+              enableOptions={enableOptions}
+              onSaveToSpotify={onSaveToSpotify}
+              onExportJSPF={onExportJSPF}
+            />
+          </Loader>
+        </div>
+      </div>
+      <BrainzPlayer
+        listens={jspfPlaylist?.playlist?.track?.map(JSPFTrackToListen) ?? []}
+        listenBrainzAPIBaseURI={APIService.APIBaseURI}
+        refreshSpotifyToken={APIService.refreshSpotifyToken}
+        refreshYoutubeToken={APIService.refreshYoutubeToken}
+      />
+    </>
+  );
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+  const { domContainer, reactProps, globalAppContext } = getPageProps();
+
+  const { user, mode, prompt, token } = reactProps;
+  const renderRoot = createRoot(domContainer!);
+  const LBRadioWithAlertNotifications = withAlertNotifications(LBRadio);
+
+  const enableOptions = token !== "";
+  renderRoot.render(
+    <ErrorBoundary>
+      <GlobalAppContext.Provider value={globalAppContext}>
+        <NiceModal.Provider>
+          <LBRadioWithAlertNotifications
+            userArg={user}
+            modeArg={mode}
+            promptArg={prompt}
+            authToken={token}
+            enableOptions={enableOptions}
```

Comment:
Since this is derived from the presence or absence of token, we don't need a separate prop for it and could instead define it directly in the `Playlist` component using the logged-in user stored in the global context:
```
const { currentUser } = useContext(GlobalAppContext);
const enableOptions = Boolean(currentUser?.auth_token);
`
``